src/transforms/onehot.jl

```diff
@@ -0,0 +1,84 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    OneHot(col)
+    
+Transforms categorical column `col` into one-hot columns of levels
+returned by the `levels` function of CategoricalArrays.jl.
+
+# Examples
+
+```julia
+OneHot(1)
+OneHot(:a)
+OneHot("a")
+```
+"""
+struct OneHot{S<:ColSelector} <: Stateless
+  col::S
+end
+
+isrevertible(::Type{<:OneHot}) = true
+
+_colname(col::Integer, names) = names[col]
+_colname(col::AbstractString, names) =
+  _colname(Symbol(col), names)
+
+function _colname(col::Symbol, names)
+  @assert col ∈ names "Invalid column selection."
+  return col
+end
+
+_name(nm, names) = 
+  nm ∈ names ? _name(Symbol("$(nm)_"), names) : nm
```

Comment:
So this function is appending an underscore when the column name already exists right? Can we get rid of it and then inline the implementation inside the apply function? You still have this tendency to create tiny isolated functions that could live in apply :)